lib/input_validation.rb

```diff
@@ -0,0 +1,64 @@
+class NilError < StandardError
```

Comment:
This should probably extend from ArgumentError.

```suggestion
class NilError < ArgumentError
```